### ISSUE TITLE
Geolocation API doesn't fit specification

### ISSUE THREAD

**Issue by EJohnF** [facebook/react-native#17232]
RN documentation says: 

> The Geolocation API extends the web spec: https://developer.mozilla.org/en-US/docs/Web/API/Geolocation

If open that site, we can find

> PositionOptions.maximumAge Secure context
Is a positive long value indicating the maximum age in milliseconds of a possible cached position that is acceptable to return. If set to 0, it means that the device cannot use a cached position and must attempt to retrieve the real current position. If set to Infinity the device must return a cached position regardless of its age. Default: 0

So that **default value is 0**

While in actual RN implementation 
[android](https://github.com/facebook/react-native/blob/1e8f3b11027fe0a7514b4fc97d0798d3c64bc895/ReactAndroid/src/main/java/com/facebook/react/modules/location/LocationModule.java#L98)
[ios](https://github.com/facebook/react-native/blob/1e8f3b11027fe0a7514b4fc97d0798d3c64bc895/Libraries/Geolocation/RCTLocationObserver.m#L62)

You can see that default value is INFINITY.

In conclusion: I think need to either edit RN docs and put an additional comment about that difference or change the implementation to fit Mozilla docs. 

Thank you!
